scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,624 @@
+        self,
+        solc_process: JsonRpcProcess,
+        test_case_name: str,
+        max_diagnostic_reports: int = 1
+    ) -> List[Any]:
+        """
+        Opens file for given test case and waits for diagnostics to be published.
+        """
+        assert max_diagnostic_reports > 0
+        solc_process.send_message(
+            'textDocument/didOpen',
+            {
+                'textDocument':
+                {
+                    'uri': self.get_test_file_uri(test_case_name),
+                    'languageId': 'Solidity',
+                    'version': 1,
+                    'text': self.get_test_file_contents(test_case_name)
+                }
+            }
+        )
+        return self.wait_for_diagnostics(solc_process, max_diagnostic_reports)
+
+    def expect_equal(self, actual, expected, description="Equality") -> None:
+        self.assertion_count = self.assertion_count + 1
+        prefix = f"[{self.assertion_count}] {SGR_ASSERT_BEGIN}{description}: "
+        diff = DeepDiff(actual, expected)
+        if len(diff) == 0:
+            self.assertions_passed = self.assertions_passed + 1
+            if self.print_assertions:
+                print(prefix + SGR_STATUS_OKAY + 'OK' + SGR_RESET)
+            return
+
+        # Failed assertions are always printed.
+        self.assertions_failed = self.assertions_failed + 1
+        print(prefix + SGR_STATUS_FAIL + 'FAILED' + SGR_RESET)
+        raise ExpectationFailed(actual, expected)
+
+    def expect_diagnostic(self, diagnostic, code: int, lineNo: int, startColumn: int, endColumn: int):
+        self.expect_equal(diagnostic['code'], code, f'diagnostic: {code}')
+        self.expect_equal(
+            diagnostic['range'],
+            {
+                'end': {'character': endColumn, 'line': lineNo},
+                'start': {'character': startColumn, 'line': lineNo}
+            },
+            "diagnostic: check range"
+        )
+    # }}}
+
+    # {{{ actual tests
+    def test_publish_diagnostics_warnings(self, solc: JsonRpcProcess) -> None:
+        self.setup_lsp(solc)
+        TEST_NAME = 'publish_diagnostics_1'
+        published_diagnostics = self.open_file_and_wait_for_diagnostics(solc, TEST_NAME)
+
+        self.expect_equal(len(published_diagnostics), 1, "One published_diagnostics message")
+        report = published_diagnostics[0]
+
+        self.expect_equal(report['uri'], self.get_test_file_uri(TEST_NAME), "Correct file URI")
+        diagnostics = report['diagnostics']
+
+        self.expect_equal(len(diagnostics), 3, "3 diagnostic messages")
+        self.expect_diagnostic(diagnostics[0], code=6321, lineNo=13, startColumn=44, endColumn=48)
+        self.expect_diagnostic(diagnostics[1], code=2072, lineNo= 7, startColumn= 8, endColumn=19)
+        self.expect_diagnostic(diagnostics[2], code=2072, lineNo=15, startColumn= 8, endColumn=20)
+
+    def test_publish_diagnostics_errors(self, solc: JsonRpcProcess) -> None:
```

Comment:
What is the difference in quality to the first test? Can you at least make this return a multi-line error?